tests/etl/test_perf_data_adapters.py

```diff
@@ -323,25 +340,144 @@ def test_no_performance_framework(test_project, test_repository,
+                          expected_subtest_alert, expected_suite_alert):
     _generate_perf_data_range(test_project, test_repository,
                               perf_option_collection, perf_platform,
-                              perf_reference_data)
+                              perf_reference_data,
+                              add_suite_value=add_suite_value,
+                              extra_suite_metadata=extra_suite_metadata,
+                              extra_subtest_metadata=extra_subtest_metadata)
+
+    expected_num_alerts = len(filter(lambda x: x is True, [expected_suite_alert,
+                                                           expected_subtest_alert]))
 
     # validate that a performance alert was generated
-    assert 1 == PerformanceAlert.objects.all().count()
-    assert 1 == PerformanceAlertSummary.objects.all().count()
+    assert expected_num_alerts == PerformanceAlert.objects.all().count()
+
+    # check # of alerts, validate summary if present
+    if expected_num_alerts > 0:
+        assert 1 == PerformanceAlertSummary.objects.all().count()
+        summary = PerformanceAlertSummary.objects.get(id=1)
+        assert summary.result_set_id == 15
+        assert summary.prev_result_set_id == 14
+    else:
+        assert 0 == PerformanceAlertSummary.objects.all().count()
+
+    # validate suite alert if it [should be] present
+    if expected_suite_alert:
+        alert = PerformanceAlert.objects.get(series_signature__test='')
+        assert alert.series_signature.suite == 'cheezburger metrics'
+        assert alert.series_signature.test == ''
+        assert alert.is_regression
+        assert alert.amount_abs == 1
+        assert alert.amount_pct == 100
+
+    # validate subtest alert if it [should be] present
+    if expected_subtest_alert:
+        alert = PerformanceAlert.objects.get(series_signature__test='test1')
+        assert alert.series_signature.suite == 'cheezburger metrics'
+        assert alert.series_signature.test == 'test1'
+        assert alert.is_regression
+        assert alert.amount_abs == 1
+        assert alert.amount_pct == 100
+
+
+def test_alert_generation_parameters(test_project, test_repository,
+                                     perf_option_collection, perf_platform,
+                                     perf_job_data, perf_reference_data):
+    # this test is mainly focused around making sure that we correctly
+    # update the series signature with the correct alert generation
+    # parameters: testing the alert generation itself happens in
+    # tests/perfalert
+    PerformanceFramework.objects.create(name="cheezburger")
+    datum = {
+        'job_guid': 'fake_job_guid',
+        'name': 'test',
+        'type': 'test',
+        'blob': {
+            'framework': {'name': "cheezburger"},
+            'suites': [
+                {
+                    'name': 'cheezburger metrics',
+                    'minBackWindow': 1,
```

Comment:
can you test with a min backwindow greater than the amount of data, thereby yielding no alert?  
I don't see tests for different thresholds.  I would like to see a generated and ignored one.